TRACING.md

```diff
@@ -0,0 +1,272 @@
+in the VM), to connect to the Jaeger agent (running on the host). The default
+Jaeger port for such communications is
+[UDP port `6831`](https://www.jaegertracing.io/docs/getting-started/).
+
+The required redirection is a achieved using two instances of the `socat(1)` network utility:
+
+- The first instance runs in the guest environment in the VM and forwards all
+  Jaeger trace traffic out of the VM using a VSOCK socket.
+
+- The second instance runs in the host environment and forwards all Jaeger
+  traffic from the VSOCK socket to the Jaeger agent's UDP port.
+
+Further details:
+
+| Name | Type | Environment | Summary | Details | Explanation |
+|-|-|-|-|-|
+| `jaeger-client-socat-redirector.service` | systemd service | Guest | Guest `UDP:6831` to guest `VSOCK:6831` | Redirects data sent to UDP port `6831` to the host context `VSOCK` socket on port `6831`. | Service that intercepts all Jaeger calls the Kata agent makes and redirects them to the Jaeger agent running in the host environment. |
+| `vsock-to-udp-server.sh` | shell script | Host | Host `VSOCK:6831` to host `UDP:6831` | Redirects all host context VSOCK data on port `6831` to the Jaeger agent, which is listening on UDP port `6831`. | Forwards all Jaeger agent calls redirected from the Kata agent to the Jaeger agent. |
+
+> **Note:**
+>
+> This is a temporary solution until Jaeger supports VSOCK natively.
+
+For tracing to work, the host system must have a Jaeger agent running.
+
+# Enabling tracing
+
+Tracing is enabled by adding the `agent.trace` option in the `kernel_params=`
+variable in the runtime's `configuration.toml` file.
+
+## Tracing options
+
+Tracing support in the agent is either disabled, or set to one of the
+following:
+
+| Trace type | Trace setting | Description | Use-case | Notes |
+|-|-|-|-|-|
+| "isolated" | `agent.trace=isolated` (or just `agent.trace`) | The traces only apply to the agent; after the container has been destroyed, the first span will start at agent startup and the last at agent shutdown | Observing agent lifespan. | |
+| "collated" | `agent.trace=collated` | In this mode, spans are associated with their `kata-runtime` initiated counterparts. | Understanding how the runtime calls the agent. | Requires runtime tracing to be enabled in `configuration.toml` (`enable_tracing=true`). |
+
+For example, to enable isolated tracing explicitly, add the following to the
+runtime's `configuration.toml` file:
+
+```
+kernel_params = "agent.trace=isolated ... "
+```
+
+> **Note:**
+>
+> Agent tracing is separate from the tracing for other Kata Containers
+> components. It is not necessary to enable runtime tracing if you want to
+> enable agent tracing (or *vice versa*). However, "collated" mode only works
+> as documented if runtime tracing is enabled.
+
+# Running the agent with tracing enabled
+
+1. Build a custom non-initrd image using
+   [osbuilder](https://github.com/kata-containers/osbuilder).
```

Comment:
Do we/can we have a better reference here - specifically stating to set `TRACE=true` in the osbuilder env/makefile for instance?

Reply:
This PR is "purely" for the agent. For what you're after, you prolly want to check out https://github.com/kata-containers/osbuilder/pull/200.